python/ray/tests/resource_isolation/test_resource_isolation_integration.py

```diff
@@ -1,44 +1,288 @@
+#
 # If you want to run this test locally, you will need to create a cgroup that
 # the raylet can manage and delegate to the correct user.
 #
-# TODO(#54703): Once implementation is complete, I will add a fixture to this
-# test to check for common errors when running locally (such as cgroup2 not mounted
-# correct). It'll follow the example of
-# src/ray/common/cgroup2/integration_tests/sysfs_cgroup_driver_integration_test_entrypoint.sh
-#
 # Run these commands locally before running the test suite:
+#
 #  sudo mkdir -p /sys/fs/cgroup/resource_isolation_test
 #  echo "+cpu +memory" | sudo tee -a /sys/fs/cgroup/resource_isolation_test/cgroup.subtree_control
 #  sudo chown -R $(whoami):$(whoami) /sys/fs/cgroup/resource_isolation_test/
 #  sudo chmod -R u+rwx /sys/fs/cgroup/resource_isolation_test/
 #  echo $$ | sudo tee /sys/fs/cgroup/resource_isolation_test/cgroup.procs
 #
 # Comment the following line out.
-_BASE_CGROUP_PATH = "/sys/fs/cgroup"
+_ROOT_CGROUP = Path("/sys/fs/cgroup")
+#
+# To run locally, uncomment the following line.
+# _ROOT_CGROUP = Path("/sys/fs/cgroup/resource_isolation_test")
+
+# The integration tests assume that the _ROOT_CGROUP exists and that
+# the process has read and write access.
+#
+# This test suite will create the following cgroup hierarchy for the tests
+# starting with BASE_CGROUP.
+#
+#                        ROOT_CGROUP
+#                             |
+#                        BASE_CGROUP
+#                       /           \
+#                 TEST_CGROUP   LEAF_CGROUP
+#                      |
+#               ray_node_<node_id>
+#               /               \
+#           system            application
+#             |                    |
+#           leaf                  leaf
 #
-# Uncomment the following line.
-# _BASE_CGROUP_PATH = "/sys/fs/cgroup/resource_isolation_test"
+# NOTE: The test suite does not assume that ROOT_CGROUP is an actual root cgroup. Therefore,
+#   1. setup will migrate all processes from the ROOT_CGROUP -> LEAF_CGROUP
+#   2. teardown will migrate all processes from the LEAF_CGROUP -> ROOT_CGROUP
+#
+# NOTE: BASE_CGROUP will have a randomly generated name to isolate tests from each other.
+#
+# The test suite assumes that
+#   1. cpu, memory controllers are available on ROOT_CGROUP i.e. in the ROOT_CGROUP/cgroup.controllers file.
+#   2. All processes inside the base_cgroup can be migrated into the leaf_cgroup to avoid not violating
+#   the no internal processes contstraint.
+#
+# All python tests should only have access to the TEST_CGROUP and nothing outside of it.
+
+_BASE_CGROUP = _ROOT_CGROUP / ("testing_" + utils.get_random_alphanumeric_string(5))
+_TEST_CGROUP = _BASE_CGROUP / "test"
+_LEAF_GROUP = _BASE_CGROUP / "leaf"
+
+_MOUNT_FILE_PATH = "/etc/mtab"
+
+# The list of processes expected to be started in the system cgroup
+# with default params for 'ray start' and 'ray.init(...)'
+_EXPECTED_SYSTEM_PROCESSES_RAY_START = [
```

Comment:
This is kind of a handy list.  Is there anything like this in the codebase or documentation that just lists it out like this?